src/NumericalAlgorithms/Spectral/SwshTags.hpp

```diff
@@ -0,0 +1,204 @@
+///   expressive code style.
+/// - `derivative_kind`: Type alias to `DerivativeKind`, represents the kind of
+///   spin-weighted derivative applied to `Tag`
+/// - `spin`: The spin weight of the scalar after the derivative has been
+///   applied.
+template <typename Tag, typename DerivativeKind>
+struct Derivative : db::PrefixTag, db::SimpleTag {
+  using type = detail::adjust_spin_weight_t<Tag, DerivativeKind>;
+  using tag = Tag;
+  // derived_from is provided as a second alias so that utilities which assume a
+  // derivative prefix tag fail earlier during compilation
+  using derived_from = Tag;
+  using derivative_kind = DerivativeKind;
+  const static int spin = type::type::spin;
+  static std::string name() noexcept {
+    return detail::compose_spin_weighted_derivative_name<DerivativeKind>(
+        Tag::name());
+  }
+};
+
+/// \ingroup SwshGroup
+/// \brief Prefix tag representing the spin-weighted spherical harmonic
+/// transform of a spin-weighted scalar.
+///
+/// Template Parameters:
+/// - `Tag`: The tag to prefix.
+///
+/// Type aliases and static values:
+/// - `type`: Always a `SpinWeighted<Scalar<ComplexModalVector>, Spin>`, where
+///   `Spin` is the same spin weight of the pre-transformed `Tag`, and 0 if
+///   provided a `Tag` with a `Type` that is not `SpinWeighted`
+/// - `tag`: An alias to the base tag `Tag`. Provided for applicability to
+///   general `db::PrefixTag` functionality
+/// - `transform_of`: Another alias to the base tag `Tag`. Provided so that
+///   utilities which use this prefix for taking transforms can have a more
+///   expressive code style.
+template <typename Tag>
+struct Coefficient : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<
+      ComplexModalVector,
+      detail::adjust_spin_weight_t<Tag, NoDerivative>::type::spin>>;
+  using tag = Tag;
+  // transform_of is provided as a second alias so that utilities which assume a
+  // derivative prefix tag fail earlier during compilation
+  using transform_of = Tag;
+  const static int spin = type::type::spin;
+  static std::string name() noexcept {
+    return "SwshTransform(" + Tag::name() + ")";
+  }
+};
+}  // namespace Tags
+
+namespace detail {
+// implementation for get_tags_with_spin
+template <typename T, typename S>
+struct has_spin : cpp17::bool_constant<T::type::type::spin == S::value> {};
+
+template <typename T, typename S>
+struct wrapped_has_spin : has_spin<typename T::tag, S> {};
+
+}  // namespace detail
+
+/// \ingroup SwshGroup
+/// \brief Extract from a provided `tmpl::list` `TagList` of tags the subset
+/// of those tags which have a static int member `spin` equal to the template
+/// parameter `Spin`. The resulting subset of `TagList` is provided in a
+/// `tmpl::list`.
+///
+/// \snippet Test_SwshTags.cpp get_tags_with_spin
+template <int Spin, typename TagList>
+using get_tags_with_spin = tmpl::remove_duplicates<tmpl::filter<
```

Comment:
do you expect to have duplicates here? Removing duplicates is a very expensive operation

Reply:
This one, yes. I expect duplicates.